src/python/pants/core/goals/fmt.py

```diff
@@ -70,14 +72,17 @@ def skipped(self) -> bool:
     def did_change(self) -> bool:
         return self.output != self.input
 
+
+@dataclass(frozen=True)
+class EnrichedFmtResult(FmtResult, EngineAwareReturnType):
```

Comment:
These "EnrichedX" classes could really use docstrings to explain why they exist.